themes/default/content/docs/esc/other-integrations/docker.md

```diff
@@ -0,0 +1,126 @@
+    AWS_SESSION_TOKEN: ${aws.login.sessionToken}
+```
+
+For the purposes of this guide, we'll create a simplified environment to demonstrate your options:
+
+```yaml
+values:
+  environmentVariables:
+    ESC_ORG: You are in the ${context.pulumi.organization.login} organization!
+    ESC_HELLO_USER: Hello, ${context.pulumi.user.login}!
+  files:
+    DOCKER_ENVFILE: |
+      ESC_ORG="You are in the ${context.pulumi.organization.login} organization!"
+      ESC_HELLO_USER="Hello, ${context.pulumi.user.login}!"
+```
+
+### Set individual environment variables in a Docker container
+
+You can [set environment variables for a Docker container](https://docs.docker.com/reference/cli/docker/container/run/#env) as part of a `docker run` command:
+
+```bash
+$ esc run <your-environment-name> -- docker run --rm -t -e ESC_ORG -e ESC_HELLO_USER alpine env
+```
+
+The output should look something like this, but with your own username and organization name set in the environment variables.
+
+```bash
+PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+HOSTNAME=e2d74889ef6d
+TERM=xterm
+ESC_HELLO_USER=Hello, example-user!
+ESC_ORG=You are in the example organization!
+HOME=/root
+```
+
+### Set environment variables from a file
+
+You can set environment variables dynamically by using an environment file as part of a `docker run` command:
+
+```bash
+$ esc run -i <your-environment-name> -- sh -c 'docker run --rm -t --env-file=$DOCKER_ENVFILE alpine env'
+```
+
+The output should look something like this, but with your own username and organization name set in the environment variables.
+
+```bash
+PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+HOSTNAME=e2d74889ef6d
+TERM=xterm
+ESC_HELLO_USER=Hello, example-user!
+ESC_ORG=You are in the example organization!
+HOME=/root
+```
+
+## Manage Docker registry credentials
+
+### Create an ESC environment with your Docker registry credentials
+
+ESC integrates with `docker` by exporting environment variables from an opened environment. You can store login configuration securely with an ESC environment.
+This example stores the username and encrypted password directly in the environment, but you can also reference external secrets with [ESC providers](/docs/esc/providers/).
+
+```yaml
+values:
+  docker:
+    username: <your-registry-username>
+    password:
+      fn::secret: <your-registry-password>
+    registry: null # Provide a registry URL if you are not using Docker Hub
+  environmentVariables:
+    USERNAME: ${docker.username}
+    PASSWORD: ${docker.password}
+    REGISTRY: ${docker.registry}
+```
+
+### Log in to a Docker registry
+
+You can [log into a Docker registry](https://docs.docker.com/reference/cli/docker/login/) without needing to manage the credentials directly in your shell:
+
+```bash
+$ esc run <your-environment-name> -- sh -c 'echo $PASSWORD | docker login --username $USERNAME --password-stdin $REGISTRY'
```

Comment:
Should also be able to do this w/o the `environmentVariables` property like so:

```console
$ esc run <your-environment-name> -- sh -c 'echo ${docker.password} | docker login --username ${docker.username} --password-stdin ${docker.registry}'
```

or

```console
$ esc run <your-environment-name> -- docker login --username \${docker.username} --password \${docker.password} \${docker.registry}
```

Reply:
That would require changing the explanation a bit, but is somewhat more succinct. If `docker` had builtin support for these envvars I'd say go that route for sure, but with command-line args I think it's a bit less clear which to direct users towards.